### PR TITLE
Improve comments output

### PR DESCRIPTION
**Pull request by Tirokk**
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-23T07:02:19Z" title="Monday, September 23rd 2019, 9:02:19 am +02:00">Sep 23, 2019</time>_
_Merged <time datetime="2019-09-23T10:45:27Z" title="Monday, September 23rd 2019, 12:45:27 pm +02:00">Sep 23, 2019</time>_
---

before, collpased:
Screenshot from 2019-09-23 09-03-00

uncollapsed:
Screenshot from 2019-09-23 09-03-10

after, collapsed:
Screenshot from 2019-09-23 09-03-31

uncollapsed:
Screenshot from 2019-09-23 09-03-39

### CODE REVIEW

File: webapp/components/Comment.vue [hunk 1]
```diff
@@ -38,14 +38,14 @@
...
-          v-if="comment.content.length < 400"
+          v-if="$filters.removeHtml(comment.content).length < 180"
           :content="comment.content"
           class="padding-left"
         />
         <div v-else class="show-more-or-less-div">
           <content-viewer
             v-if="isCollapsed"
-            :content="$filters.truncate(comment.content, 400)"
+            :content="$filters.truncate(comment.content, 180)"
```
Comment (Tirokk) [inline webapp/components/Comment.vue:48]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-23T10:18:08Z" title="Monday, September 23rd 2019, 12:18:08 pm +02:00">Sep 23, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -38,14 +38,14 @@
       </div>
       <div v-show="!openEditCommentMenu">
         <content-viewer
-          v-if="comment.content.length < 400"
+          v-if="$filters.removeHtml(comment.content).length < 180"
           :content="comment.content"
           class="padding-left"
         />
         <div v-else class="show-more-or-less-div">
           <content-viewer
             v-if="isCollapsed"
-            :content="$filters.truncate(comment.content, 400)"
+            :content="$filters.truncate(comment.content, 180)"
```


Please refactor the 180 to be not such a magic number. You could put it into `constants/comments.js`.

File: webapp/package.json [hunk 2]
```diff
@@ -77,6 +77,7 @@
     "tippy.js": "^4.3.5",
     "tiptap": "~1.25.0",
     "tiptap-extensions": "~1.27.0",
+    "trunc-html": "^1.1.2",
```
Comment (Tirokk) [inline webapp/package.json:80]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-23T10:18:32Z" title="Monday, September 23rd 2019, 12:18:32 pm +02:00">Sep 23, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -77,6 +77,7 @@
     "tippy.js": "^4.3.5",
     "tiptap": "~1.25.0",
     "tiptap-extensions": "~1.27.0",
+    "trunc-html": "^1.1.2",
```


Very good. You chose the one which has more Github stars than the other two options.

Comment (Tirokk) [reply webapp/package.json:80]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-23T10:25:11Z" title="Monday, September 23rd 2019, 12:25:11 pm +02:00">Sep 23, 2019</time>_

---


hehe, I chose the one we use in the `excerptMiddleware`, but whoever chose it must have looked at the GitHub stars :smile_cat:

File: webapp/components/Comment.vue (hunk 1 above)
Comment (Tirokk) [reply webapp/components/Comment.vue:48]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-09-23T10:33:49Z" title="Monday, September 23rd 2019, 12:33:49 pm +02:00">Sep 23, 2019</time>_

---


I agree with @roschaefer . 😍

File: webapp/plugins/vue-filters.js [hunk 3]
```diff
@@ -45,9 +46,9 @@ export default ({ app = {} }) => {
       if (length <= 0) {
         return value
       }
-      let output = value.substring(0, length)
+      let output = trunc(value, length).html
       if (output.length < value.length) {
-        output += '…'
+        output += ' …'
```
Comment (Tirokk) [inline webapp/plugins/vue-filters.js:51]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-09-23T10:34:37Z" title="Monday, September 23rd 2019, 12:34:37 pm +02:00">Sep 23, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -45,9 +46,9 @@ export default ({ app = {} }) => {
       if (length <= 0) {
         return value
       }
-      let output = value.substring(0, length)
+      let output = trunc(value, length).html
       if (output.length < value.length) {
-        output += '…'
+        output += ' …'
```


I really love this @mattwr18 ! 💫😍
The ellipse the most and for sure the space as well. 💥

**Review by Mogge — APPROVED** [review]
Great! You can refactor it a little but it already gets my approval.

approved

In the future I would put this implementation on the server maybe and stop saving `contentExcerpt` to the database. (Given, that it does not lead to performance problems)

**Review by Mogge — APPROVED** [review]
I approve it like it is. 😍💫

PS: My additional suggestion, in case you change the code at all, would be:
After Update a comment switch to `show less` otherwise there are ugly effects in displaying the content.

Girl with thumbs up

**Pull request closed without merging**